### PR TITLE
Add subscriptions to RPC interface

### PR DESCRIPTION
**Pull request by nanocryk**

### CODE REVIEW

File: client/orchestrator-chain-rpc-interface/src/lib.rs [hunk 1]
```diff
@@ -247,20 +259,26 @@ impl OrchestratorChainInterface for OrchestratorChainRpcClient {
...
+        let stream = tokio_stream::wrappers::ReceiverStream::new(rx);
+        Ok(stream.boxed())
     }
 
     /// Get a stream of finality notifications.
     async fn finality_notification_stream(
         &self,
     ) -> OrchestratorChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
-        todo!()
+        let rx = self.send_register_message(WsClientRequest::RegisterFinalizationListener)?;
```
Comment (girazoki) [inline client/orchestrator-chain-rpc-interface/src/lib.rs:280]: why do we need this register messagE?

Comment (girazoki) [reply client/orchestrator-chain-rpc-interface/src/lib.rs:280]: ah got it, I also see it in cumulus

**Review by girazoki — APPROVED** [review]
I see it pretty similar to cumulus, hence my approval. @tmpolaczyk can you take a look?

**Review by tmpolaczyk — APPROVED** [review]
I didn't try it out but looks good

**Pull request merged**